src/erlcloud_s3.erl

```diff
@@ -1042,10 +1042,18 @@ s3_request(Config, Method, Host, Path, Subreasource, Params, POSTData, Headers)
 
 %% s3_request2 returns {ok, Body} or {error, Reason} instead of throwing as s3_request does
 %% This is the preferred pattern for new APIs
-s3_request2(Config, Method, Host, Path, Subresource, Params, POSTData, Headers) ->
+s3_request2(Config, Method, Bucket, Path, Subresource, Params, POSTData, Headers) ->
     case erlcloud_aws:update_config(Config) of
         {ok, Config1} ->
-            s3_request4_no_update(Config1, Method, Host, Path, Subresource, Params, POSTData, Headers);
+            Response = s3_request4_no_update(Config1, Method, Bucket, Path, 
+                   Subresource, Params, POSTData, Headers),
+            case Config1#aws_config.s3_follow_redirect of
```

Comment:
may i ask to put this case inside the `s3_follow_redirect ` function and under the error case match. 
so that 
s3_follow_redirect
     case request of
           error and follow
                redirect
           _ -> response